comp/dogstatsd/server/server.go

```diff
@@ -816,23 +817,35 @@ func (s *server) parseServiceCheckMessage(parser *parser, message []byte, origin
 	return serviceCheck, nil
 }
 
+// getBuckets converts the buckets from the configuration into slice of float64. The default from the configuration are
+// slice of strings
 func getBuckets(cfg model.Reader, logger log.Component, option string) []float64 {
-	if !cfg.IsSet(option) {
-		return nil
-	}
-
-	buckets, err := cfg.GetFloat64SliceE(option)
-	if err != nil {
-		logger.Errorf("%s, falling back to default values", err)
-		return nil
-	}
+	buckets, _ := getBucketsError(cfg, logger, option)
 	if len(buckets) == 0 {
 		logger.Debugf("'%s' is empty, falling back to default values", option)
 		return nil
 	}
 	return buckets
 }
 
+func getBucketsError(cfg model.Reader, logger log.Component, option string) ([]float64, error) {
+	if !cfg.IsSet(option) {
+		return nil, fmt.Errorf("unknown setting %s", option)
```

Comment:
This error won't be handled by the caller, it will just be ignored. Under normal circumstances that shouldn't be possible, but I can imagine future changes to this code making this into a bug. Might be better to leave the original error handling in the caller and log.Error there.